com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -442,14 +443,15 @@ define([
                 $scope.errors.newStartDate = [];
                 $scope.errors.newEndDate = [];
 
-                $scope.validateDates($scope.edit_data.new_role_id.newStartDate, $scope.edit_data.new_role_id.newEndDate, function (error, field) {
-                    errors++;
-                    if (field.indexOf('start_date') > -1) {
-                        $scope.errors.newStartDate.push(error);
-                    }
-                    if (field.indexOf('end_date') > -1) {
-                        $scope.errors.newEndDate.push(error);
-                    }
+                var contract = me.contractsData[$scope.edit_data.new_role_id.job_contract_id];
+
+                errors = validateDates({
```

Comment:
I think the signature for `validateDates()` could be better, as now you're passing a single object containing both the dates that it needs to validate and the errors recipients. Maybe separating the two could make it more readable:
```js
/**
 *
 * @param {object} dates - The dates to validate
 * @param {object} errors - The error recipients
 */
function validateDates(dates, errors) { /* .. */ }

// Usage
validateDate(
    {
        'start': startDate,
        'end': endDate,
        'contractStart': contractStartDate,
        'contractEnd': contractEndDate
    },
    {
        'start': errorStart,
        'end': errorEnd
    }
));